flytekit/image_spec/image_spec.py

```diff
@@ -82,6 +83,7 @@ class ImageSpec:
     platform: str = "linux/amd64"
     pip_index: Optional[str] = None
     pip_extra_index_url: Optional[List[str]] = None
+    pip_extra_args: Optional[List[str]] = None
```

Comment:
Can we make this a single string and not a list?

```suggestion
    pip_extra_args: Optional[str] = None
```

Reply:
I made the update to take a space-delimited string. What's the rationale behind this? `pip_install_args` is joined with " " later, so taking a list and extending the list seems more aligned with how it is implemented. 

For example, when adding multiple command line arguments, it is done by extending the list of multiple elements, not a single element with multiple command line arguments concatenated with spaces.
https://github.com/flyteorg/flytekit/blob/5a2c44f1976665c420541d3385bf946cf959b2f6/flytekit/image_spec/default_builder.py#L195

(a separate topic but it may be worth using `shlex` instead of `" ".join`)

Reply:
I see that using a list is more aligned with the implementation.

I was mostly thinking about the developer experience. If one wants to add many arguments, they can use a single string:

```python
image = ImageSpec(
    pip_extra_args="--prerelease --all-extras --compile-bytecode --only-binary"
)
```

It takes a little more work to make it a list of strings:

```python
image = ImageSpec(
    pip_extra_args=["--prerelease", "--all-extras", "--compile-bytecode", "--only-binary"]
)
```

Do you have a preference when it comes to developer experience?

Reply:
Good point. Given that these args are not parsed, a single string is simpler and less confusing.